pyiron_atomistics/sphinx/output_parser.py

```diff
@@ -386,3 +387,97 @@ def get_convergence(self):
     def get_fermi(self):
         pattern = r"Fermi energy:\s+(\d+\.\d+)\s+eV"
         return np.array(re.findall(pattern, self.log_main)).astype(float)
+
+
+class SphinxWavesParser:
+    """Class to read SPHInX waves.sxb files (HDF5 format)
+
+    Initialize with waves.sxb filename, or use load ()
+
+    """
+
+    def __init__(self, file_name="waves.sxb", cwd="."):
+        """
+        Args:
+            file_name (str): file name
+            cwd (str): directory path
+        """
+        self._eps = None
+        if Path(file_name).is_absolute():
+            self.wfile = h5py.File(Path(file_name))
+        else:
+            path = Path(cwd) / Path(file_name)
+            self.wfile = h5py.File(path)
+
+    @property
+    def _n_gk(self):
+        return self.wfile["nGk"][:]
+
+    @property
+    def _fft_idx(self):
+        fft_idx = []
+        off = 0
+        for ngk in self._n_gk:
+            fft_idx.append(self.wfile["fftIdx"][off : off + ngk])
+            off += ngk
+        return fft_idx
+
+    @property
+    def mesh(self):
+        return self.wfile["meshDim"][:]
+
+    @property
+    def Nx(self):
+        return self.wfile["meshDim"][0]
+
+    @property
+    def Ny(self):
+        return self.wfile["meshDim"][1]
+
+    @property
+    def Nz(self):
+        return self.wfile["meshDim"][2]
+
+    @property
+    def n_states(self):
+        return self.wfile["nPerK"][0]
+
+    @property
+    def n_spin(self):
+        return self.wfile["nSpin"].shape[0]
+
+    @property
+    def k_weights(self):
+        return self.wfile["kWeights"][:]
+
+    @property
+    def k_vec(self):
+        return self.wfile["kVec"][:]
+
+    @property
+    def eps(self):
+        """All eigenvalues (in Hartree) as (nk,n_states) block"""
+        if self._eps is None:
+            self._eps = self.wfile["eps"][:].reshape(-1, self.n_spin, self.n_states)
+        return self._eps.T  # change
+
+    def get_psi_rec(self, i, ispin, ik):
+        """Loads a single wavefunction on full FFT mesh"""
+        if i < 0 or i >= self.n_states:
+            raise IndexError(f"i={i} fails 0 <= i < n_states={self.n_states}")
+        if ispin < 0 or ispin >= self.n_spin:
+            raise IndexError(f"ispin={ispin} fails 0 <= ispin < n_spin={self.n_spin}")
+        if ik < 0 or ik >= self.nk:
+            raise IndexError(f"ik={ik} fails 0 <= ik < nk={self.nk}")
+
+        res = np.zeros(shape=self.mesh, dtype=np.complex128)
+        off = self._n_gk[ik] * (i + ispin * self.n_states)
+        psire = self.wfile[f"psi-{ik+1}.re"][off : off + self._n_gk[ik]]
+        psiim = self.wfile[f"psi-{ik+1}.im"][off : off + self._n_gk[ik]]
```

Comment:
The indices are not the same as the ones used in `_fft_idx`? Can't we simply use `self.wfile["fftIdx"][off : off + ngk]` instead of `_fft_idx[ik]`? @freyso

Reply:
Sorry for the late response - they are not the same.
There is 1 fft_idx per k-point, stored contiguously in the file. The original version loaded this into a python list of numpy vectors. Very efficient appending, to answer the concern above. This was loaded once from file and then reused for all psi (which are many per k-point, and `off` refers to the index within the k-specific psi-variable inside the file).
The current version is wrong due to index problems of the fft_idx.

Reply:
ok if the data is stored it might be efficient, but that's not what's happening in the current implementation, so we have to change `_fft_idx` in the way that it's stored in a variable.